**Assignment Chapter 1 - Foundation for Data Science\
Hastialisna Hurul Aeni Setiawan - SC62421**

# Nomor 1

## Membaca dan Menampilkan DataFrame





In [ ]:
import pandas as pd

sheet_url = 'https://docs.google.com/spreadsheets/d/1g974wLSP9ariF3-kzONoRg1kPyl-2na6/edit#gid=1975587336'
sheet_url_csv=sheet_url.replace('/edit#gid=','/export?format=csv&gid=') #Convert to CSV
print(sheet_url_csv)
df_24=pd.read_csv(sheet_url_csv)
df_24.head(5)

https://docs.google.com/spreadsheets/d/1g974wLSP9ariF3-kzONoRg1kPyl-2na6/export?format=csv&gid=1975587336


No. Peserta                  Nama             Jurusan  Berat Badan (kg)  \
0     SC62401    Alvin Brain Sinaga           Sosiologi                55   
1     SC62419     Anindyta Shayna W          Statistika                48   
2     SC62412    Ardian Dwi Refanda    Sistem Informasi                74   
3     SC62402       Bagus Sulistiyo  Teknik Informatika                65   
4     SC62420  Dea Shafira Anindita          Statistika                65   

   Tinggi Badan (cm)    Domisili   Akun Instagram  
0                173   Pekanbaru    alvin_brain18  
1                155    Semarang        sshaynaaa  
2                179  Bojonegoro     ardianfanda_  
3                179   Tangerang  bagussulistiyoo  
4                160  Yogyakarta         deashfra

## Informasi tentang DataFrame

In [ ]:
#Menampilkan informasi DataFrame
print("\033[1mBanyak kolom dan baris serta jenis data:\033[0m")
df_24.info()

print("\n\033[1mStatistika deskriptif dari Berat badan dan Tinggi badan:\033[0m")
df_24.describe()

Banyak kolom dan baris serta jenis data:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24 entries, 0 to 23
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   No. Peserta        24 non-null     object
 1   Nama               24 non-null     object
 2   Jurusan            24 non-null     object
 3   Berat Badan (kg)   24 non-null     int64 
 4   Tinggi Badan (cm)  24 non-null     int64 
 5   Domisili           24 non-null     object
 6   Akun Instagram     24 non-null     object
dtypes: int64(2), object(5)
memory usage: 1.4+ KB

Statistika deskriptif dari Berat badan dan Tinggi badan:


Berat Badan (kg)  Tinggi Badan (cm)
count         24.000000          24.000000
mean          55.500000         164.291667
std           12.607796           9.429275
min           39.000000         150.000000
25%           45.000000         156.500000
50%           53.000000         163.500000
75%           65.000000         170.250000
max           81.000000         185.000000

In [ ]:
#Menghitung Jumlah tiap Domisili
domisili_count = df_24["Domisili"].value_counts()
print("\033[1mRekap domisili\033[0m:\n")
print(domisili_count)

Rekap domisili:

Surabaya      2
Nganjuk       2
Purwokerto    2
Bandung       2
Pekanbaru     1
Jepara        1
Bandung       1
Jambi         1
Pati          1
Pasuruan      1
Bali          1
Baturaja      1
Jakarta       1
Semarang      1
Bekasi        1
Manado        1
Yogyakarta    1
Tangerang     1
Bojonegoro    1
Serang        1
Name: Domisili, dtype: int64


In [ ]:
#Menghitung Jumlah tiap Jurusan
jurusan_count = df_24["Jurusan"].value_counts()
print("\033[1mRekap jurusan\033[0m:\n")
print(jurusan_count)

Rekap jurusan:

Sistem Informasi             5
Teknik Informatika           5
Matematika                   3
Statistika                   2
Pendidikan Bahasa Inggris    2
Matematika                   2
Manajemen                    2
Sosiologi                    1
Ilmu Komputer                1
Teknik Industri              1
Name: Jurusan, dtype: int64


## Soal 1

a. Identifikasi dan penjelasan tipe data yang ada pada dataframe \\
*   No.Peserta, Nama, Jurusan, Domisili, Akun Instagram: String
*   Berat badan, Tinggi badan: Integer

b. Banyak kolom dan baris
*   Terdapat 7 kolom dan 24 baris

c. Informasi siswa kompi 24
* Terdapat 24 siswa  
* Rata-rata berat badan dan tinggi badan berturut-turut adalah 55.5 kg dan tinggi 164.291667 cm
*   Berat badan paling besar yaitu 81 kg dan paling kecil yaitu 39 kg
*   Tinggi badan paling tinggi yaitu 185 cm dan paling kecil yaitu 150 kg
*   Domisili sangat bervariasi tetapi terdapat 4 domisili yang terbanyak ditempati, yaitu Surabaya, Nganjuk, Purwokerto, dan Bandung
*   Jurusan paling banyak adalah Sistem Informasi dan Teknik Informatika masing-masing sebanyak 5 siswa, sedangkan jurusan paling sedikit adalah Ilmu Komputer, Sosiologi, dan Teknik Industri masing-masing sebanyak 1 siswa

## Soal 2

BMI Tiap Siswa Kompi 24

In [ ]:
#Membuat variabel baru tinggi badan dengan convert cm ke m
tinggi_badan = df_24["Tinggi Badan (cm)"] / 100

#Menghitung BMI
BMI = df_24["Berat Badan (kg)"] / (tinggi_badan ** 2)

print(BMI)

0     18.376825
1     19.979188
2     23.095409
3     20.286508
4     25.390625
5     15.943878
6     17.333333
7     18.730489
8     18.507766
9     18.314256
10    18.991965
11    18.289895
12    19.959355
13    19.031142
14    24.163265
15    27.700831
16    19.562955
17    27.181630
18    20.202020
19    27.309968
20    17.008821
21    18.025957
22    19.382145
23    16.880194
dtype: float64


## Soal 3

Membuat Program BMI

In [ ]:
InputTinggiBadan = int(input("Masukan Tinggi Badan : "))
InputBeratBadan = int(input("Masukan Berat Badan : "))

ResultBmi = InputBeratBadan / ((InputTinggiBadan / 100) ** 2);
print(ResultBmi)

## Soal 4

Update dataframe df dengan menambahkan hasil perhitungan soal nomor 2, beri nama kolom BMI.

In [ ]:
df_24["BMI"]=BMI
df_24.head()

No. Peserta                  Nama             Jurusan  Berat Badan (kg)  \
0     SC62401    Alvin Brain Sinaga           Sosiologi                55   
1     SC62419     Anindyta Shayna W          Statistika                48   
2     SC62412    Ardian Dwi Refanda    Sistem Informasi                74   
3     SC62402       Bagus Sulistiyo  Teknik Informatika                65   
4     SC62420  Dea Shafira Anindita          Statistika                65   

   Tinggi Badan (cm)    Domisili   Akun Instagram        BMI  
0                173   Pekanbaru    alvin_brain18  18.376825  
1                155    Semarang        sshaynaaa  19.979188  
2                179  Bojonegoro     ardianfanda_  23.095409  
3                179   Tangerang  bagussulistiyoo  20.286508  
4                160  Yogyakarta         deashfra  25.390625

## Soal 5

Buatlah status BMI untuk setiap siswa dari kategori berikut, lalu simpan data statusnya dengan nama kolom Status BMI.
* Normal: 18.5-24.9
* Overweight: 25.0-29.9
* Obesity: 30.0-34.9
* Underweight: < 18.5


In [ ]:
def status(bmi):
  if bmi > 30 :
    return "Obesity"
  elif bmi > 25 :
    return "Overweight"
  elif bmi > 18.5 :
    return "Normal"
  else:
    return "Underweight"

for bmi in df_24["BMI"]:
  print(status(bmi))

Underweight
Normal
Normal
Normal
Overweight
Underweight
Underweight
Normal
Normal
Underweight
Normal
Underweight
Normal
Normal
Normal
Overweight
Normal
Overweight
Normal
Overweight
Underweight
Underweight
Normal
Underweight


In [ ]:
df_24["BMI Status"] = df_24["BMI"].apply(status)
df_24.head()

No. Peserta                  Nama             Jurusan  Berat Badan (kg)  \
0     SC62401    Alvin Brain Sinaga           Sosiologi                55   
1     SC62419     Anindyta Shayna W          Statistika                48   
2     SC62412    Ardian Dwi Refanda    Sistem Informasi                74   
3     SC62402       Bagus Sulistiyo  Teknik Informatika                65   
4     SC62420  Dea Shafira Anindita          Statistika                65   

   Tinggi Badan (cm)    Domisili   Akun Instagram        BMI   BMI Status  
0                173   Pekanbaru    alvin_brain18  18.376825  Underweight  
1                155    Semarang        sshaynaaa  19.979188       Normal  
2                179  Bojonegoro     ardianfanda_  23.095409       Normal  
3                179   Tangerang  bagussulistiyoo  20.286508       Normal  
4                160  Yogyakarta         deashfra  25.390625   Overweight

## Soal 6

a. Buatlah variabel baru lalu urutkan data berdasarkan tinggi badan dari yang tertinggi ke rendah.\
b. Buat variabel baru lalu urutkan data berdasarkan Nama dari Z -> A.

In [ ]:
#Sort dari tinggi tertinggi ke terendah
df_24_sort=df_24
df_24_sort.sort_values(by=['Tinggi Badan (cm)'], ascending=False).head()

No. No. Peserta                        Nama             Jurusan  \
10   11     SC62405  Joshua Roy Danna Lacanlale  Teknik Informatika   
2     3     SC62412          Ardian Dwi Refanda    Sistem Informasi   
3     4     SC62402             Bagus Sulistiyo  Teknik Informatika   
14   15     SC62407     Muhammad Naufal Baihaqi  Teknik Informatika   
0     1     SC62401          Alvin Brain Sinaga           Sosiologi   

    Berat Badan (kg)  Tinggi Badan (cm)    Domisili       Akun Instagram  \
10                65                185      Bekasi  darkfateashseigneur   
2                 74                179  Bojonegoro         ardianfanda_   
3                 65                179   Tangerang      bagussulistiyoo   
14                74                175    Surabaya           naufalqii_   
0                 55                173   Pekanbaru        alvin_brain18   

          BMI   BMI Status  
10  18.991965       Normal  
2   23.095409       Normal  
3   20.286508       Normal  
14  24.163265       Normal  
0   18.376825  Underweight

In [ ]:
#Sort dari nama Z ke A
df_24_sortname=df_24
df_24_sortname.sort_values(by=['Nama'], ascending=False).head()

No. No. Peserta                 Nama           Jurusan  Berat Badan (kg)  \
23   24     SC62411       Yuli Fitriyani         Manajemen                39   
22   23     SC62418       Winola Farhani         Manajemen                49   
21   22     SC62424  Suci Permata Kusuma        Matematika                45   
20   21     SC62410   Siti Luna Satiyah        Matematika                 43   
19   20     SC62417       Reysha Pratiwi  Sistem Informasi                78   

    Tinggi Badan (cm)  Domisili Akun Instagram        BMI   BMI Status  
23                152   Bandung  yuli.ftriyani  16.880194  Underweight  
22                159    Serang          wfhni  19.382145       Normal  
21                158    Jepara  sucikusuma27_  18.025957  Underweight  
20                159  Bandung        lunassa_  17.008821  Underweight  
19                169     Jambi       reyshap_  27.309968   Overweight

## Soal 7

Filter dataframe df berdasarkan:\
a. Status BMI di luar Normal.\
b. Tinggi Badan > 169 cm dan Berat Badan > 60 kg\
c. Nama yang mengandung huruf “y”.\
d. Siapa saja yang berat badannya kurang dari 60 kg?


In [ ]:
#Status BMI di luar Normal
filtered_a = df_24[['Nama', 'BMI Status']][df_24['BMI Status'] != 'Normal']
filtered_a

Nama   BMI Status
0                  Alvin Brain Sinaga  Underweight
4                Dea Shafira Anindita   Overweight
5                  Donna Arianti Dewi  Underweight
6               Elisa Windyan Maranti  Underweight
9               Ilma Hamala Nur Faiza  Underweight
11               Mei Dwi Wuryaningsih  Underweight
15        Ngakan Made Alit Wiradhanta   Overweight
17  Perwira Annissa Dyah Permatasari    Overweight
19                     Reysha Pratiwi   Overweight
20                 Siti Luna Satiyah   Underweight
21                Suci Permata Kusuma  Underweight
23                     Yuli Fitriyani  Underweight

In [ ]:
#Tinggi Badan > 169 cm dan Berat Badan > 60 kg
filtered_b = df_24[['Nama','Tinggi Badan (cm)', 'Berat Badan (kg)']][(df_24['Tinggi Badan (cm)'] > 169) & (df_24['Berat Badan (kg)'] > 60)]
filtered_b

Nama  Tinggi Badan (cm)  Berat Badan (kg)
2            Ardian Dwi Refanda                179                74
3               Bagus Sulistiyo                179                65
10   Joshua Roy Danna Lacanlale                185                65
14      Muhammad Naufal Baihaqi                175                74
15  Ngakan Made Alit Wiradhanta                171                81

In [ ]:
#Nama mengandung huruf "y"
filtered_c = df_24[['Nama']] [df_24['Nama'].str.contains("y", case=False)]
filtered_c

Nama
1                   Anindyta Shayna W
3                     Bagus Sulistiyo
6               Elisa Windyan Maranti
7         Frasty Julistyfenia Rambing
10         Joshua Roy Danna Lacanlale
11               Mei Dwi Wuryaningsih
17  Perwira Annissa Dyah Permatasari 
19                     Reysha Pratiwi
20                 Siti Luna Satiyah 
23                     Yuli Fitriyani

In [ ]:
#Berat badan kurang dari 30 kg
filtered_d = df_24[['Nama', 'Berat Badan (kg)']][df_24['Berat Badan (kg)'] < 60]
filtered_d

Nama  Berat Badan (kg)
0                 Alvin Brain Sinaga                55
1                  Anindyta Shayna W                48
5                 Donna Arianti Dewi                45
6              Elisa Windyan Maranti                39
7        Frasty Julistyfenia Rambing                45
8   Hastialisna Hurul Aeni Setiawan                 51
9              Ilma Hamala Nur Faiza                44
11              Mei Dwi Wuryaningsih                48
12         Melfiza Puspita Maharishi                55
13                     Muhamad Akmal                55
16                       Nur Azizah                 47
18          Rasthania Silva Fadlila                 55
20                Siti Luna Satiyah                 43
21               Suci Permata Kusuma                45
22                    Winola Farhani                49
23                    Yuli Fitriyani                39

## Soal 8

Tambahkan nomor urut pada dataframe df sesuai jumlah siswa dan tempatkan kolom nomor baru tersebut di paling ujung kiri sebagai kolom pertama serta beri nama kolomnya “No.”!

In [ ]:
df_24_new=df_24
df_24_new.insert(0, "No.", range(1, len(df_24_new)+1))
df_24_new

No. No. Peserta                               Nama  \
0     1     SC62401                 Alvin Brain Sinaga   
1     2     SC62419                  Anindyta Shayna W   
2     3     SC62412                 Ardian Dwi Refanda   
3     4     SC62402                    Bagus Sulistiyo   
4     5     SC62420               Dea Shafira Anindita   
5     6     SC62403                 Donna Arianti Dewi   
6     7     SC62413              Elisa Windyan Maranti   
7     8     SC62404        Frasty Julistyfenia Rambing   
8     9     SC62421   Hastialisna Hurul Aeni Setiawan    
9    10     SC62414              Ilma Hamala Nur Faiza   
10   11     SC62405         Joshua Roy Danna Lacanlale   
11   12     SC62415               Mei Dwi Wuryaningsih   
12   13     SC62406          Melfiza Puspita Maharishi   
13   14     SC62422                      Muhamad Akmal   
14   15     SC62407            Muhammad Naufal Baihaqi   
15   16     SC62416        Ngakan Made Alit Wiradhanta   
16   17     SC62408                        Nur Azizah    
17   18     SC62423  Perwira Annissa Dyah Permatasari    
18   19     SC62409           Rasthania Silva Fadlila    
19   20     SC62417                     Reysha Pratiwi   
20   21     SC62410                 Siti Luna Satiyah    
21   22     SC62424                Suci Permata Kusuma   
22   23     SC62418                     Winola Farhani   
23   24     SC62411                     Yuli Fitriyani   

                      Jurusan  Berat Badan (kg)  Tinggi Badan (cm)  \
0                   Sosiologi                55                173   
1                  Statistika                48                155   
2            Sistem Informasi                74                179   
3          Teknik Informatika                65                179   
4                  Statistika                65                160   
5   Pendidikan Bahasa Inggris                45                168   
6            Sistem Informasi                39                150   
7          Teknik Informatika                45                155   
8                  Matematika                51                166   
9            Sistem Informasi                44                155   
10         Teknik Informatika                65                185   
11                 Matematika                48                162   
12  Pendidikan Bahasa Inggris                55                166   
13              Ilmu Komputer                55                170   
14         Teknik Informatika                74                175   
15         Teknik Informatika                81                171   
16           Teknik Industri                 47                155   
17           Sistem Informasi                67                157   
18                Matematika                 55                165   
19           Sistem Informasi                78                169   
20                Matematika                 43                159   
21                 Matematika                45                158   
22                  Manajemen                49                159   
23                  Manajemen                39                152   

      Domisili       Akun Instagram        BMI   BMI Status  
0    Pekanbaru        alvin_brain18  18.376825  Underweight  
1     Semarang            sshaynaaa  19.979188       Normal  
2   Bojonegoro         ardianfanda_  23.095409       Normal  
3    Tangerang      bagussulistiyoo  20.286508       Normal  
4   Yogyakarta             deashfra  25.390625   Overweight  
5      Nganjuk    donnaarianti_dewi  15.943878  Underweight  
6   Purwokerto              _elsjnr  17.333333  Underweight  
7       Manado        frastyrambing  18.730489       Normal  
8      Bandung       hastialisnahas  18.507766       Normal  
9      Nganjuk              ilmaml_  18.314256  Underweight  
10      Bekasi  darkfateashseigneur  18.991965       Normal  
11  Purwokerto           meiiyaaa__  18.289895  Underweigh

## Soal 9

Cari nilai-nilai berikut:\
a. Cari nilai terendah dan tertingginya.\
b. Berapa nilai rata-rata tinggi badan siswa?\
c. Berapa rata-rata berat badan siswa?\
d. Ada berapa orang jumlah siswa  yang tercatat pada dataframe df?\
e. 5 orang dengan nilai BMI tertinggi.\
f. Ada berapa orang yang tergolong status Underweight, Normal, Overweight, dan Obesity?\


In [ ]:
#Nilai terendah, tertinggi, rata-rata, jumlah siswa
df_24.describe()

No.  Berat Badan (kg)  Tinggi Badan (cm)        BMI
count  24.000000         24.000000          24.000000  24.000000
mean   12.500000         55.500000         164.291667  20.402018
std     7.071068         12.607796           9.429275   3.473132
min     1.000000         39.000000         150.000000  15.943878
25%     6.750000         45.000000         156.500000  18.308166
50%    12.500000         53.000000         163.500000  19.206643
75%    18.250000         65.000000         170.250000  20.988733
max    24.000000         81.000000         185.000000  27.700831

In [ ]:
#5 Orang dengan nilai BMI tertinggi
df_24.nlargest(5,"BMI")

No. No. Peserta                               Nama             Jurusan  \
15   16     SC62416        Ngakan Made Alit Wiradhanta  Teknik Informatika   
19   20     SC62417                     Reysha Pratiwi    Sistem Informasi   
17   18     SC62423  Perwira Annissa Dyah Permatasari     Sistem Informasi   
4     5     SC62420               Dea Shafira Anindita          Statistika   
14   15     SC62407            Muhammad Naufal Baihaqi  Teknik Informatika   

    Berat Badan (kg)  Tinggi Badan (cm)    Domisili  Akun Instagram  \
15                81                171        Bali  its.alitssss\n   
19                78                169       Jambi        reyshap_   
17                67                157    Surabaya      annissadyh   
4                 65                160  Yogyakarta        deashfra   
14                74                175    Surabaya      naufalqii_   

          BMI  BMI Status  
15  27.700831  Overweight  
19  27.309968  Overweight  
17  27.181630  Overweight  
4   25.390625  Overweight  
14  24.163265      Normal

In [ ]:
#Hitung siswa sesuai status BMI
statusbmi_count = df_24["BMI Status"].value_counts()
print("\033[1mRekap status\033[0m:\n")
statusbmi_count

Rekap status:



Normal         12
Underweight     8
Overweight      4
Name: BMI Status, dtype: int64

# Nomor 2

Perusahaan Investasi Jamin-Sugih Co kami memberikan penawaran investasi terbaik dengan pembagian keuntungan tetap sebesar 25% setiap tahun.

Buatlah program Python yang dapat menerima input dari pengguna dalam bentuk:\
Modal Investasi (bilangan bulat antara 0-1.000.000.000)\
Jangka Waktu Investasi (bilangan bulat antara 1-25 dalam tahun)\
Selain itu, program akan menampilkan error message bahwa modal atau tenure tidak sesuai ketentuan.

\
Program akan menampilkan informasi akumulasi investasi setiap tahun, menggunakan iterasi untuk setiap tahun.

Contoh:\
Input
> Money Capital : 100000\
> Investment Tenure (years) : 2

Output
> Year 1, your total investment : 125000.0\
> Year 2, your total investment : 156250.0


In [ ]:
from tabulate import tabulate

def Investasi(modal, jangka_waktu):
    # Cek apakah modal dan jangka waktu sesuai ketentuan
    if not 0 < modal <= 1000000000:
        print("\033[94mError: Modal investasi harus dalam rentang antara 1 hingga 1.000.000.000\033[0m")
        return
    if not 0 < jangka_waktu <= 25:
        print("\033[94mError: Jangka waktu investasi harus dalam rentang antara 1 hingga 25 tahun\033[0m")
        return

    # Menghitung keuntungan setiap tahun dan menampilkan informasi dalam tabel
    table = []
    total_modal = modal
    for tahun in range(1, jangka_waktu + 1):
        keuntungan_tahunan = total_modal * 0.25
        total_modal += keuntungan_tahunan
        table.append([tahun, modal, keuntungan_tahunan, total_modal])

    # Tampilkan tabel dengan judul berwarna biru
    headers = ["Tahun", "Modal Awal", "Keuntungan", "Total Modal"]
    print("\033[94m" + tabulate(table, headers=headers, tablefmt="pretty") + "\033[0m")

# Input dari pengguna diluar fungsi investasi
modal_investasi = int(input("Masukkan modal investasi (antara 1 hingga 1.000.000.000): "))
jangka_waktu_investasi = int(input("Masukkan jangka waktu investasi (antara 1 hingga 25 tahun): "))

# Memanggil fungsi investasi dengan input pengguna
Investasi(modal_investasi, jangka_waktu_investasi)